src/objdump.main.v3

```diff
@@ -59,5 +62,44 @@ def parseAndDump(engine: Engine, path: string) -> int {
 		if (f.imp == null) OUT.put1(" { %d bytes }", f.orig_bytecode.length);
 		OUT.outln();
 	}
+	for (i < module.heaptypes.length) {
+		var h = module.heaptypes[i];
+		OUT.put1("heaptypes[%d] = ", i);
+		h.render(OUT);
+		var fh = h.getFirstSuperType();
+		if (fh != null) fh.render(OUT);
+		OUT.outln();
+	}
+	for (i < module.abstypes.length) {
+		var a = module.abstypes[i];
+		OUT.put1("abstypes[%d] = ", i);
+		OUT.put1(": %z", a.isAssignableTo(ValueType.I32));
+		OUT.outln();
+	}
+	for (i < module.tables.length) {
+		var t = module.tables[i];
+		OUT.put1("tables[%d] = ", i);
+		OUT.put1(": has_default_elem %z", t.has_default_elem);
+		OUT.outln();
+	}
+	for (i < module.tags.length) {
+		var t = module.tags[i];
+		OUT.put1("tags[%d] = ", i);
+		if (t.fields != null) {
+			for (value in t.fields) {
+				OUT.put1("value: %q", value.render);
+				OUT.outln();
+			}
+		}
+		OUT.outln();
+	}
+	for (i < module.exports.length) {
+		var e = module.exports[i];
+		OUT.put1("exports[%d] = ", i);
```

Comment:
How about printing out what kind of declaration and that declaration's index here?

Reply:
I'm a bit confused about how to get the declaration type and index from Decl. I only saw the following two options
```
var imp: ImportInfo;
def imported() -> bool { return imp != null; }
```
And the ImportInfo class is empty. I tried using ".index" but it didn't work.

Reply:
Oh right. The exports array has a string and a `Decl`. You can check the type of the decl to see what it is, e.g.

```
match (e.1) {
  x: FuncDecl => ...
  x: MemoryDecl => ...
  x: TableDecl => ...
  x: GlobalDecl => ...
}
```